### PR TITLE
monitoring: Migrate core-application alerts to Cloud SaaS and DevOps teams

### PR DESCRIPTION
**Pull request by indradhanush**
Cloud SaaS team is here in OpsGenie: https://sourcegraph.app.opsgenie.com/teams/dashboard/0b6de43a-2eff-4e78-aa19-698c39c13312/main.

DevOps team is here in OpsGenie: https://sourcegraph.app.opsgenie.com/teams/dashboard/9ec2825d-38da-4e2b-bdec-a0c03d11d420/main

### CODE REVIEW

**Comment by unknwon** [comment]
The current official name and case for Cloud SaaS is "Cloud SaaS" (the ending "S" is not in upper case) 😁 and using hyphen seems fine IMO, so "cloud-saas".

**Comment by indradhanush** [comment]
@unknwon The difficulty with using `cloud-saas` is that the generated handbook link then points to https://about.sourcegraph.com/handbook/engineering/cloud-saas, which is a 404. :disappointed: 

Any suggestions how we overcome this limitation?

**Comment by quinnkeast** [comment]
> @unknwon The difficulty with using `cloud-saas` is that the generated handbook link then points to https://about.sourcegraph.com/handbook/engineering/cloud-saas, which is a 404. 😞
> 
> Any suggestions how we overcome this limitation?

We could just add a new page at that destination linking to the correct page (not sure if there's a way to set up an auto  redirect... @jeanduplessis do you know?).

**Comment by jeanduplessis** [comment]
> We could just add a new page at that destination linking to the correct page (not sure if there's a way to set up an auto redirect... @jeanduplessis do you know?).

In the handbook we automatically redirect based on git history. If you need to inject a manual redirect you can do so in `src/scripts/generate-redirects.mjs` in the handbook project.

**Comment by quinnkeast** [comment]
> > We could just add a new page at that destination linking to the correct page (not sure if there's a way to set up an auto redirect... @jeanduplessis do you know?).
> 
> In the handbook we automatically redirect based on git history. If you need to inject a manual redirect you can do so in `src/scripts/generate-redirects.mjs` in the handbook project.

Thanks! For this case, is this a best practice or a bit of an antipattern we should avoid in favour of just creating a page?

(I've had to maintain a gigantic redirects file before... I imagine we don't want to end up in that scenario with the handbook.)

**Comment by jeanduplessis** [comment]
> Thanks! For this case, is this a best practice or a bit of an antipattern we should avoid in favour of just creating a page?

As far as possible we'd want to avoid manual redirects. If its not too much trouble I suggest rather adding a page in the handbook.

**Comment by quinnkeast** [comment]
@indradhanush addressed in [this PR](https://github.com/sourcegraph/handbook/pull/608). At the very least, it won't 404 now.

File: doc/admin/observability/alert_solutions.md [hunk 1]
```diff
@@ -179,7 +179,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
 ]
 ```
 
-<sub>*Managed by the [Sourcegraph Core application team](https://about.sourcegraph.com/handbook/engineering/core-application).*</sub>
+<sub>*Managed by the [Sourcegraph Cloud-saas team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:182]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 2]
```diff
@@ -204,7 +204,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:207]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 3]
```diff
@@ -553,7 +553,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:556]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 4]
```diff
@@ -674,7 +674,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:677]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 5]
```diff
@@ -699,7 +699,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:702]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 6]
```diff
@@ -724,7 +724,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:727]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 7]
```diff
@@ -749,7 +749,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:752]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 8]
```diff
@@ -774,7 +774,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:777]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 9]
```diff
@@ -799,7 +799,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:802]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 10]
```diff
@@ -824,7 +824,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:827]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 11]
```diff
@@ -847,7 +847,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:850]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 12]
```diff
@@ -870,7 +870,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:873]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 13]
```diff
@@ -893,7 +893,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:896]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 14]
```diff
@@ -1167,7 +1167,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1170]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 15]
```diff
@@ -1603,7 +1603,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1606]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 16]
```diff
@@ -1628,7 +1628,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1631]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 17]
```diff
@@ -1651,7 +1651,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1654]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 18]
```diff
@@ -1675,7 +1675,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1678]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 19]
```diff
@@ -1699,7 +1699,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1702]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 20]
```diff
@@ -1723,7 +1723,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1726]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 21]
```diff
@@ -1748,7 +1748,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1751]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 22]
```diff
@@ -1773,7 +1773,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1776]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 23]
```diff
@@ -1798,7 +1798,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1801]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 24]
```diff
@@ -1823,7 +1823,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1826]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 25]
```diff
@@ -1846,7 +1846,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1849]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 26]
```diff
@@ -1903,7 +1903,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:1906]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 27]
```diff
@@ -2539,7 +2539,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:2542]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: doc/admin/observability/alert_solutions.md [hunk 28]
```diff
@@ -3554,7 +3554,7 @@ with your code hosts connections or networking issues affecting communication wi
...
```
Comment (quinnkeast) [inline doc/admin/observability/alert_solutions.md:3557]:
```suggestion
<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://about.sourcegraph.com/handbook/engineering/cloud-saas).*</sub>
```

File: monitoring/definitions/frontend.go [hunk 29]
```diff
@@ -117,7 +117,7 @@ func Frontend() *monitoring.Container {
 							Query:       `histogram_quantile(0.9, sum by(le) (rate(src_http_request_duration_seconds_bucket{route="blob"}[10m])))`,
 							Critical:    monitoring.Alert().GreaterOrEqual(5, nil),
 							Panel:       monitoring.Panel().LegendFormat("latency").Unit(monitoring.Seconds),
-							Owner:       monitoring.ObservableOwnerCoreApplication,
+							Owner:       monitoring.ObservableOwnerCloudSaaS,
```
Comment (indradhanush) [inline monitoring/definitions/frontend.go:120]: Blob latency should go under repo management as it deals with the /blob route which ultimately hits gitserver.

Note for myself to fix tomorrow.

**Comment by bobheadxi** [comment]
@indradhanush you can customize the team names/links generation here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@ee6d8e9894360ac743f52bbcde54a7a0db696282/-/blob/monitoring/monitoring/monitoring.go?L399:26

**Comment by unknwon** [comment]
@indradhanush From my understanding and based on the conversation in the Slack, I think Cloud SaaS wants to go through each of these alerts and make sure we're equipped knowledge to handle in case of alerts fired. Which would effectively blocking this PR to be merged anytime soon, is that correct?

File: doc/admin/observability/alert_solutions.md [hunk 30]
```diff
@@ -3555,7 +3555,7 @@ with your code hosts connections or networking issues affecting communication wi
 ]
 ```
 
-<sub>*Managed by the [Sourcegraph Core application team](https://handbook.sourcegraph.com/engineering/core-application).*</sub>
+<sub>*Managed by the [Sourcegraph Cloud SaaS team](https://handbook.sourcegraph.com/engineering/cloud/saas).*</sub>
```
Comment (rafax) [inline doc/admin/observability/alert_solutions.md:3558]: This one should likely stay with repo management ?

**Review by rafax — COMMENTED** [review]
I tried to understand the alerts being changed, and I think a few of them should have other (non-Cloud-SaaS) owners, but LMK what you think

File: doc/admin/observability/alert_solutions.md [hunk 31]
```diff
@@ -1904,7 +1904,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (rafax) [inline doc/admin/observability/alert_solutions.md:1907]: This is for precise-code-intel - should it be owned by code-intel?

File: doc/admin/observability/alert_solutions.md [hunk 32]
```diff
@@ -2540,7 +2540,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (rafax) [inline doc/admin/observability/alert_solutions.md:2543]: This is for worker - should this be owned by code-intel?

**Comment by daxmc99** [comment]
Some of these disputed alerts can also be owned by the [devops](https://sourcegraph.app.opsgenie.com/teams/dashboard/9ec2825d-38da-4e2b-bdec-a0c03d11d420/main) team

File: doc/admin/observability/alert_solutions.md [hunk 33]
```diff
@@ -1168,7 +1168,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
[the remark this answers is not in the document]
Comment (indradhanush) [reply doc/admin/observability/alert_solutions.md:1171]: Agreed. Will remove from this PR.

File: doc/admin/observability/alert_solutions.md [hunk 34]
```diff
@@ -1803,7 +1803,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (indradhanush) [inline doc/admin/observability/alert_solutions.md:1806]: @daxmc99 Should these be owned by the devops team as well?

File: doc/admin/observability/alert_solutions.md [hunk 35]
```diff
@@ -1753,7 +1753,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (indradhanush) [inline doc/admin/observability/alert_solutions.md:1756]: Same for this and other DB related alerts?

File: doc/admin/observability/alert_solutions.md [hunk 36]
```diff
@@ -1633,7 +1633,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (kopancek) [inline doc/admin/observability/alert_solutions.md:1636]: Should this be owned by cloud dev-ops instead?

File: doc/admin/observability/alert_solutions.md [hunk 37]
```diff
@@ -1681,7 +1681,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (kopancek) [inline doc/admin/observability/alert_solutions.md:1684]: Same as above

File: doc/admin/observability/alert_solutions.md [hunk 38]
```diff
@@ -1705,7 +1705,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (kopancek) [inline doc/admin/observability/alert_solutions.md:1708]: Same as above

File: doc/admin/observability/alert_solutions.md [hunk 39]
```diff
@@ -1729,7 +1729,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (kopancek) [inline doc/admin/observability/alert_solutions.md:1732]: Same as above

File: doc/admin/observability/alert_solutions.md [hunk 40]
```diff
@@ -1778,7 +1778,7 @@ To learn more about Sourcegraph's alerting and how to set up alerts, see [our al
...
```
Comment (kopancek) [inline doc/admin/observability/alert_solutions.md:1781]: Same as above

**Review by kopancek — COMMENTED** [review]
Suggested ownership change to postgres.

Not sure about the API requests and Frontend latency, maybe we want to keep it on cloud-saas side. My hunch is we will probably need to ping devops for these most of the time as well.

**Review by rafax — APPROVED** [review]
LGTM on the 2 alerts owned by Cloud SaaS

**Review by daxmc99 — APPROVED** [review]
LGTM on alerts for @sourcegraph/cloud-devops 
FYI some of these may currently be silenced

**Pull request merged**